src/transport/zerotier/zerotier.c

```diff
@@ -2706,13 +2710,13 @@ zt_ep_set_add_local_addr(void *arg, const void *data, size_t sz, nni_type t)
 		case NNG_AF_INET:
 			sin                  = (void *) &ss;
 			sin->sin_family      = AF_INET;
+			sin->sin_port        = sa.s_in.sa_port;
```

Comment:
My decision *not* to fix the local port was intentional.  Do you have a need for this?

Reply:
Yes, this one I need. The local addresses that are communicated to peers must have both the ip and port (so the peers can try to directly contact at that ip/port). This is why I also made a method to retrieve the random selected udp port that is created for the zt udp transport.

Reply:
By informing peers of the local addresses to try, there is a higher success of direct connections instead of being relayed by the zerotier root servers, especially if the peers are behind the same nat firewall.

Reply:
Ok, I may have totally misunderstood what we are trying to do.

The "local address" I thought to mean the local address on the system -- meaning our *own* IP address.  The value zero was meant to mean that the system should attempt to bind to any available port, not fix to a single hard coded port.

If the local address in ZeroTier means something different, like the address of a peer on the same subnet, that's a completely different thing, and the original code was bad.

It would be good to get full clarification here.

Reply:
You are talking about the wildcard listen address of the udp transport, which is set up in zt_node_create. 

The zt_ep_set_add_local_addr function only informs the zerotier node / transport of the different ip addresses that peers can use to reach the node. (I have in my broker program a function that enumerates all the ip address active on the host, and calls zt_ep_set_add_local_addr).

Reply:
So zt_ep_set_add_local_addr is called with the different IPs on the host, with the port used for the wildcard listen.

Reply:
I retrieve the port with zt_ep_get_udp4_addr before I call zt_ep_set_add_local_addr.

Reply:
See for example: https://github.com/capitar/capitar-zerotier-broker/blob/393e2d01d4c441bbd3036e05b919614238ffdeb5/util.c#L321

Reply:
Ok, that helps clarify it for me.  I do wonder if it would be better to use a single option to get both the v4 and v6 addresses -- we pretty much don't make any such distinction anywhere else in the API.

Reply:
Yes, that make sense. I'll update the pull request with that.

Reply:
This more complicated than I thought. It is not possible to have both an ipv6 and and an ipv4 address / port in nng_sockaddr, as it is a union of these.

So I will have to seek another data type to return, as I cannot use nng_sockaddr for this.

Perhaps keep it just as it is?

Reply:
For completeness I changed it to also store the ip address, not only the port.

Reply:
I may change this somewhat... it's clear that what I was confused about this when I did the original work.

My intention for address retrieval was that this would return an array (just two) of nng_sockaddrs, not overload a single one.

Reply:
Sounds good. Thanks!